rust/kernel/sync/mod.rs

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Synchronisation primitives.
+//!
+//! This module contains the kernel APIs related to synchronisation that have been ported or
+//! wrapped for usage by Rust code in the kernel and is shared by all of them.
+//!
+//! # Example
+//!
+//!```
+//! fn test() {
```

Comment:
No need for wrapping into a `fn test() {` in examples, i.e. the code can go directly in "file" scope.